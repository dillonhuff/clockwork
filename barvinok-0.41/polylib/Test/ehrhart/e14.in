# Philippe's simple example
#--------------------------
# 1 <= i <= N
# 1 <= j <= N
# 1 <= k <= N
# i+j+k-2 = M
#--------------------------
7 7
#  i  j  k  N  M  1
1  1  0  0  0  0 -1
1 -1  0  0  1  0  0
1  0  1  0  0  0 -1
1  0 -1  0  1  0  0
1  0  0  1  0  0 -1
1  0  0 -1  1  0  0
0 -1 -1 -1  0  1  2
#--------------------------
0 4

N M
Philippe's simple example

#--------------------------
# I have done the work by hand, and here is the right
# answer :
#
# If 1 <= M <= N then EP(N,M) = 1/2 M^2 + 1/2 M
# If N <= M <= 2N-1 then EP(N,M) = -3/2 N^2 - M^2 + 3NM - M
# If 2N-1 <= M <= 3N-2 then EP(N,M) = 9/2 N^2 + 1/2 M^2 - 3NM - 3/2 N + 1/2 M
