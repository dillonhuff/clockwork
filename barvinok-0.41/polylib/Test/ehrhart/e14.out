---------------------------------------
Domain:
         -2N + M + 1 >= 0
         3N - M  -2 >= 0
          1 >= 0

Vertices:
[ -2N+M+2,  N,  N ]
[ N,  -2N+M+2,  N ]
[ N,  N,  -2N+M+2 ]

Ehrhart Polynomial:
( 9/2 * N^2 + ( -3 * M + -3/2 )
 * N + ( 1/2 * M^2 + 1/2 * M + 0 )
 )

---------------------------------------
Domain:
         2N - M  -1 >= 0
         - N + M  >= 0
          1 >= 0

Vertices:
[ -N+M+1,  N,  1 ]
[ -N+M+1,  1,  N ]
[ N,  -N+M+1,  1 ]
[ N,  1,  -N+M+1 ]
[ 1,  -N+M+1,  N ]
[ 1,  N,  -N+M+1 ]

Ehrhart Polynomial:
( -3/2 * N^2 + ( 3 * M + 3/2 )
 * N + ( -1 * M^2 + -1 * M + 0 )
 )

---------------------------------------
Domain:
         N - M  >= 0
         M  -1 >= 0
          1 >= 0

Vertices:
[ M,  1,  1 ]
[ 1,  M,  1 ]
[ 1,  1,  M ]

Ehrhart Polynomial:
( 1/2 * M^2 + 1/2 * M + 0 )

